target/linux/realtek/dts-5.10/rtl8380_tplink_sg2xxx.dtsi

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "rtl838x.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x10000000>;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	gpio-restart {
+		compatible = "gpio-restart";
+		gpios = <&gpio0 3 GPIO_ACTIVE_LOW>;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: led-0 {
+			label = "green:power";
+			gpios = <&gpio0 13 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	i2c1: i2c-gpio-1 {
+		compatible = "i2c-gpio";
+		scl-gpios = <&gpio0 0 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+		sda-gpios = <&gpio0 1 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+		i2c-gpio,delay-us = <2>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		tps23861@30 {
+			compatible = "ti,tps23861";
+			reg = <0x30>;
+			shunt-resistor-micro-ohms = <255000>;
+		};
```

Comment:
If I leave the nodes, with `status = "disabled"`, then the SG2008P gets in a bootloop. I'm not going to be debugging this:

```
[   11.250000] rtl83xx-switch switch@1b000000: configuring for fixed/internal link mode
[   11.260000] DSA: tree 0 setup
[   11.260000] LINK state irq: 20
[   11.260000] In rtl83xx_setup_qos
[   11.270000] Setting up RTL838X QoS
[   11.270000] RTL838X_PRI_SEL_TBL_CTRL(i): 00033112
[   11.280000] Current Intprio2queue setting: 00000000
[   11.280000] QM_PKT2CPU_INTPRI_MAP: 00fac688
[   11.290000] rtl838x_dbgfs_init called
[   11.300000] CPU 0 Unable to handle kernel paging request at virtual address 00000000, epc == 802c7b58, ra == 8022f8cc
[   11.310000] Oops[#1]:
[   11.310000] CPU: 0 PID: 893 Comm: kworker/0:3 Not tainted 5.10.138 #0
[   11.310000] Workqueue: events deferred_probe_work_func
[   11.310000] $ 0   : 00000000 00000001 00000000 8216cf60
[   11.310000] $ 4   : 00000000 00000000 811fd654 82084000
[   11.310000] $ 8   : 00000000 00000001 811f0000 127a3980
[   11.310000] $12   : 00000000 0000d190 00000000 00000100
[   11.310000] $16   : fffffffe 8306e440 00000000 81200000
[   11.310000] $20   : 81200000 821a4bc4 00000003 fffffffe
[   11.310000] $24   : 00000000 989c4992                  
[   11.310000] $28   : 8265a000 8265bc20 80710000 8022f8cc
[   11.310000] Hi    : 00000000
[   11.310000] Lo    : 000003c0
[   11.310000] epc   : 802c7b58 strlen+0x0/0x2c
[   11.310000] ra    : 8022f8cc start_creating.part.0+0x78/0x194
[   11.310000] Status: 1010fc03 KERNEL EXL IE 
[   11.310000] Cause : 10800008 (ExcCode 02)
[   11.310000] BadVA : 00000000
[   11.310000] PrId  : 00019070 (MIPS 4KEc)
[   11.310000] Modules linked in:
[   11.310000] Process kworker/0:3 (pid: 893, threadinfo=(ptrval), task=(ptrval), tls=00000000)
[   11.310000] Stack : 00000002 83079708 83077d60 00000000 821a4080 8306e440 00000000 00000060
[   11.310000]         00000018 802301ac 0000003c 0000000f 806a0000 80230634 83077898 00000000
[   11.310000]         00000000 806a0000 821a4080 8306e440 821a4080 803a4494 00000003 fffffffe
[   11.310000]         80710000 80230248 806a21ec 8066e0bc 821a4080 8306e440 00000018 821a4464
[   11.310000]         821a4bb8 821a4bc4 00000003 fffffffe 80710000 803a4ba0 82087700 806a13a8
[   11.310000]         ...
[   11.310000] Call Trace:
[   11.310000] [<802c7b58>] strlen+0x0/0x2c
[   11.310000] [<8022f8cc>] start_creating.part.0+0x78/0x194
[   11.310000] [<802301ac>] debugfs_create_dir+0x1c/0x15c
[   11.310000] [<803a4494>] rtl838x_dbgfs_port_init+0x54/0x258
[   11.310000] [<803a4ba0>] rtl838x_dbgfs_init+0xe0/0x56c
[   11.310000] [<807b9a7c>] rtl83xx_sw_probe+0x7d4/0x81c
[   11.310000] [<80328b7c>] platform_drv_probe+0x40/0x94
[   11.310000] [<80326448>] really_probe+0x124/0x55c
[   11.310000] [<80323d3c>] bus_for_each_drv+0x70/0xb0
[   11.310000] [<80326990>] __device_attach+0x110/0x1f0
[   11.310000] [<80325144>] bus_probe_device+0x9c/0xb8
[   11.310000] [<80325764>] deferred_probe_work_func+0xbc/0x114
[   11.310000] [<80040aa0>] process_one_work+0x1f0/0x460
[   11.310000] [<80041060>] worker_thread+0x350/0x5c4
[   11.310000] [<800473b4>] kthread+0x13c/0x144
[   11.310000] [<80001bb8>] ret_from_kernel_thread+0x14/0x1c
[   11.310000] 
[   11.310000] Code: a066ffff  1000fff7  00000000 <80820000> 10400007  00000000  00801025  80430001  1460fffe 
[   11.310000] 
[   11.570000] ---[ end trace 12da47b96c41a0b4 ]---
[   11.580000] Kernel panic - not syncing: Fatal exception
[   11.580000] Rebooting in 1 seconds..
```

Reply:
It seems that disabling the port (not phy) will cause the crash:

```
&port24 {
	status = "disabled";
};
```

I would like to model the board layout such that:
- the board is described in the dtsi
- unpopulated footprints are described in the device dts

I am going to leave the `/delete-node/` for now, unless you tell me (again) otherwise. I think this approach is the least amount of code duplication. I hope you agree.

Reply:
The crash should be fixed with 9a7f17e11f5d ("realtek: ignore disabled switch ports").

If you have two 8-port devices and one 10-port devices you can support with this dtsi, then enabled those two ports by default sounds a bit backwards to me. I think using `/delete-node/` twice for sounds like more confusing than using `status = "disabled"` to indicate it's not populated.